Gems/Pointcloud/Code/Source/Clients/PointcloudComponent.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "PointcloudComponent.h"
+#include <3rd/happly.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Pointcloud/PointcloudTypeIds.h>
+#include <Render/PointcloudFeatureProcessor.h>
+namespace Pointcloud
+{
+
+    PointcloudComponent::PointcloudComponent(const AZ::Data::Asset<PointcloudAsset>& pointcloudAsset, const float pointSize)
+        : m_pointcloudAsset(pointcloudAsset)
+        , m_pointSize(pointSize)
+    {
+    }
+
+    void PointcloudComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<PointcloudComponent, AZ::Component>()
+                ->Version(0)
+                ->Field("PointcloudAsset", &PointcloudComponent::m_pointcloudAsset)
+                ->Field("PointSize", &PointcloudComponent::m_pointSize);
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<PointcloudComponent>("PointcloudComponent", "PointcloudComponent")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "PointcloudComponent")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "RobotecTools")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PointcloudComponent::m_pointcloudAsset,
+                        "Pointcloud Asset",
+                        "Asset containing the pointcloud data")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PointcloudComponent::m_pointSize,
+                        "Point Size",
+                        "Size of the points in the pointcloud");
```

Comment:
If we are also providing `PointcloudEditorComponent` This part should be there.
Now there are 2 pointcloud components, only difference is if they show pointcloud in editor. 
![image](https://github.com/user-attachments/assets/bb7bd31d-50c7-4f5d-996e-1bc93725d26a)

Reply:
Removed this info